### DISCUSSION TITLE
Filter Legends

### DISCUSSION THREAD

**Discussion by Ghaydn** [DFHack/dfhack#4262] [Ideas]
Currently available tools like Legends Browser are less informative and useful than in-game legends mode. But since DFHack already can filter certain things in the game, it might be not very complicated to add similar search and filter to the legends mode. Filter events by type and search by text - is it possible?

Examples:
1. I want to read a civilisation's history. But it is full of competitions and performances, so I'll have to list through that mess to find something interesting. If I only could hide these events...
2. A necromancer became a king and I want to find out when he learned secrets of life and death. Dude has long biography, so it's hard to find a single string in that mess. Text search could help here.
3. Someone waves schemes in a civilisation. All schemes have similar coloring and similar length, so it's hard to find all schemes of this particular intriguer. Text search could help here too.
4. Age chronicles is the only place where wars and battles are shown. It would be very helpful to be able to find wars by involved civilisations, even though entries like **in early spring of 105, Vesifa Sesi, "The Conflict of Shooting" began.** don't contain civilisation names by themselves.

Legends mode is a huge part of DF and it is usally underrated, because of how uncomfortable it is to use this mode. Search and filtering would make reading Legends much easier and more accessible.

**Comment by myk002** [comment]
Whether something is filterable depends on the data structure that is being used to drive the display list. If the list is backed directly by the source of truth, then filtering the list means removing data from the source of truth. This can have other repercussions, such as incomplete exports if you choose to export to xml while the list is filtered.

I'm not sure if this applies here - I haven't looked at the data structures involved - but it's something to keep in mind.

The second potential complication is that the lists can get **huge**. Any filtering might need to happen in c++ for performance reasons, which may make legends mode filtering logic a bit more complex than existing search and sort.

The third thing to check is where features like this appear on Putnam's roadmap. If she's planning on addressing this in vanilla soon (for some value of "soon"), we shouldn't duplicate that work.
